packages/firebase_ui_firestore/lib/src/query_builder.dart

```diff
@@ -470,14 +478,45 @@ class FirestoreListView<Document> extends FirestoreQueryBuilder<Document> {
               return emptyBuilder(context);
             }
 
+            final itemCount = snapshot.docs.length;
+
             return ListView.builder(
-              itemCount: snapshot.docs.length,
+              itemCount: itemCount,
               itemBuilder: (context, index) {
-                final isLastItem = index + 1 == snapshot.docs.length;
-                if (isLastItem && snapshot.hasMore) snapshot.fetchMore();
+                final isLastItem = index + 1 == itemCount;
+                if (!showFetchingIndicator && isLastItem && snapshot.hasMore) {
+                  snapshot.fetchMore();
+                }
 
                 final doc = snapshot.docs[index];
-                return itemBuilder(context, doc);
+                return showFetchingIndicator
+                    ? OnMountListener(
+                        onMount: () {
+                          if (isLastItem && snapshot.hasMore) {
+                            snapshot.fetchMore();
+                          }
+                        },
+                        child: Column(
+                          crossAxisAlignment: CrossAxisAlignment.start,
+                          children: [
+                            itemBuilder(context, doc),
+                            if (isLastItem && snapshot.hasMore)
+                              fetchingIndicatorBuilder?.call(context) ??
+                                  const Padding(
+                                    padding: EdgeInsets.symmetric(
+                                      vertical: 16.0,
+                                    ),
+                                    child: Center(
+                                      child: LoadingIndicator(
+                                        size: 30.0,
+                                        borderWidth: 2.0,
+                                      ),
```

Comment:
Could we make all these configurable via constructor properties with default values?

Reply:
We have already provided the user with an option to use fetchingIndicatorBuilder if they don't like the default implementation. Therefore, it doesn't seem necessary to clutter the constructor with these properties.